### DISCUSSION TITLE
Category: Short Description (PLEASE CHANGE)

### DISCUSSION THREAD

**Discussion by FitzM** [photoprism/photoprism#3488] [Q&A]
PLEASE ONLY PROCEED IF YOU ARE ABSOLUTELY SURE THAT THIS IS NOT A TECHNICAL SUPPORT INCIDENT AND/OR A PROBLEM WITH OTHER SOFTWARE YOU ARE USING. PLEASE ALSO CHECK OUR PUBLIC ROADMAP AND TRY TO FIND EXISTING FEATURE REQUESTS FIRST:

- <https://link.photoprism.app/roadmap>
- <https://github.com/photoprism/photoprism/issues>
 
VISIT <https://www.photoprism.app/kb/getting-support> TO LEARN MORE ABOUT OUR SUPPORT OPTIONS. THANK YOU FOR YOUR CAREFUL CONSIDERATION!

===============================================================================

**Describe what problem this solves and why this would be valuable to many users**

- The NSFW filtering feature often catches a lot of items that aren't NSFW. Is there a way to train the model based on what it's identified? These images are right,  these are wrong?

**Comment by graciousgrey** [comment]
We agree that there is room for improvement.

At the moment it is not possible to train the NSFW model. Adding the functionality to train the NSFW or image classification model might be theoretically possible. However, we probably won't get to that in the near future.

See also: 
- https://github.com/photoprism/photoprism/issues/448

**Reply by FitzM** [reply]
Can you point me to the direction of the model and source of the testing data? I might be able to figure out something for training it.

**Reply by lastzero** [reply]
We use an existing TensorFlow model, which we didn't train ourselves. You can probably find details about the model in our public code (package `internal/nsfw`) and somewhere in the Developer Guide or on [dl.photoprism.app](https://dl.photoprism.app/), where the model may be available for download.

Note that I implemented this a long time ago with the primary goal of stopping spam on our demo instance. So depending on what specific files and use case you have, it may not meet your needs. Of course, you are welcome to provide us with a better model! :)
